experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -365,17 +378,19 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	/**
 	 * Spawn a new branch that is based off of the current state of this branch.
 	 * Changes made to the new branch will not be applied to this branch until the new branch is merged back in.
+	 * @param repairDataStoreProvider - a RepairDataStoreProvider that reflects the state of the new branch
 	 * @param anchors - an optional set of anchors that the new branch is responsible for rebasing
 	 */
 	public fork(
-		repairDataStoreProvider?: IRepairDataStoreProvider,
+		repairDataStoreProvider: IRepairDataStoreProvider,
```

Comment:
It's odd, from an API perspective, that forking a branch requires passing this in. After all, the branch already has one, and it can be cloned. Can you try to remove it? At the very least, I think we can make it optional again, and if it is not provided, then we call `clone` on the existing one. The only reason that we can't do that in every case is because when `SharedTreeView` forks itself, it also needs references to the forest and schema... perhaps there is even a way around that, though it's not obvious to me at first glance.

Reply:
One way we would get around this is to expose the forest from the RepairDataStoreProvider and grab it from there rather than have it cloned in the view. It does make ownership boundaries kinda gross though.

Reply:
I think it's totally fair to have `ForestRepairDataStoreProvider` expose the forest and the schema, but we can't expose it from `IRepairDataStoreProvider` of course (because it might not even have one). And when forking a view, you'd first fork the branch (which would clone the provider), and then you'd have to grab the provider out of the forked branch and use its forest and schema to populate the forked view. That's all fine and I actually think that's the best way to do this, except that the branch exposes `IRepairDataStoreProvider`, not `ForestRepairDataStoreProvider`, so you'd need to make branch generic over the type of provider, or something. Which maybe isn't that bad, but I can't be sure without actually trying it.

Reply:
I'm guessing a lot of the grossness is radiating outward from ForestRepairDataStore's need read from the forest. There's a good chance the ForestRepairDataStore will stop needing access to the forest soon. It all depends on whether we can ditch support for the SetValue API. I'm not really following this discussion but if there's a way to keep this grossness contained it may be better to do since it might go away, even if that means a less safe API surface.

Reply:
My vote now is to just keep the optional. I briefly tried the approach we talked about and it was pretty gross.